src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsnonassociationsapi/service/NonAssociationsService.kt

```diff
@@ -29,11 +39,46 @@ class NonAssociationsService(
     return nonAssociationsRepository.findById(id).getOrNull()?.toDto()
   }
 
-  fun getDetails(prisonerNumber: String): LegacyNonAssociationDetails {
+  fun getPrisonerNonAssociations(prisonerNumber: String, options: NonAssociationOptions): PrisonerNonAssociations {
+    val nonAssociations = nonAssociationsRepository.findAllByFirstPrisonerNumber(prisonerNumber) +
+      nonAssociationsRepository.findAllBySecondPrisonerNumber(prisonerNumber)
+
+    val prisonerNumbers = (
+      listOf(prisonerNumber) +
+        nonAssociations.map(NonAssociationJPA::firstPrisonerNumber) +
+        nonAssociations.map(NonAssociationJPA::secondPrisonerNumber)
+      ).distinct()
+    val prisoners = offenderSearch.searchByPrisonerNumbers(prisonerNumbers)
+
+    var nonAssociationsFiltered = nonAssociations
+    // filter out non-associations in other prisons
+    if (options.onlySamePrison) {
+      val prisonId = prisoners[prisonerNumber]!!.prisonId
+      nonAssociationsFiltered = nonAssociationsFiltered.filter { nonna ->
+        prisoners[nonna.firstPrisonerNumber]!!.prisonId == prisonId &&
+          prisoners[nonna.secondPrisonerNumber]!!.prisonId == prisonId
+      }
+    }
+    // filter out closed non-associations
+    if (!options.includeClosed) {
+      nonAssociationsFiltered = nonAssociationsFiltered.filter { nonna ->
+        !nonna.isClosed
+      }
+    }
+
+    return nonAssociationsFiltered.toPrisonerNonAssociations(prisonerNumber, prisoners)
+  }
+
+  fun getLegacyDetails(prisonerNumber: String): LegacyNonAssociationDetails {
     return prisonApiService.getNonAssociationDetails(prisonerNumber)
   }
 
   private fun persistNonAssociation(nonAssociation: NonAssociationJPA): NonAssociationJPA {
     return nonAssociationsRepository.save(nonAssociation)
   }
 }
+
+data class NonAssociationOptions(
```

Comment:
could the name be something like _filter_ options? cos it's not options configuring a non-association itself

Reply:
I didn't name it "filter" 'cause I left it open for configuring more things, e.g. sorting, pagination, etc...but I will rename to plural as it makes more sense.

Reply:
(and I wouldn't really want to have multiple parameters passed in for different options related to the non-associations you're getting back)

Reply:
https://github.com/ministryofjustice/hmpps-non-associations-api/pull/35/commits/7945067d480f0cde81ddbf697f02def473d9455b